install/installer/pkg/components/image-builder-mk3/tlssecret.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package image_builder_mk3
+
+import (
+	"fmt"
+
+	"github.com/gitpod-io/gitpod/installer/pkg/common"
+	"github.com/gitpod-io/gitpod/installer/pkg/config/v1"
+
+	certmanagerv1 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func tlssecret(ctx *common.RenderContext) ([]runtime.Object, error) {
+	// Only enable TLS in workspace clusters. This check can be removed
+	// once image-builder-mk3 has been removed from application clusters
+	// (https://github.com/gitpod-io/gitpod/issues/7845).
+	if ctx.Config.Kind != config.InstallationWorkspace {
+		return nil, nil
+	}
+
+	serverAltNames := []string{
+		fmt.Sprintf("%s.%s.svc", Component, ctx.Namespace),
+		fmt.Sprintf("%s.%s.svc.cluster.local", Component, ctx.Namespace),
+		Component,
+		fmt.Sprintf("%s-dev", Component),
```

Comment:
Is this dedicated to the Harvester preview environment?

Reply:
Yes, `ws-dev` is for harvester previews

Context:
https://github.com/gitpod-io/gitpod/blob/4aacec9e264892232d76aba227f02e6f33da13f5/dev/preview/infrastructure/harvester/dns.tf#L32

https://github.com/gitpod-io/gitpod/blob/4aacec9e264892232d76aba227f02e6f33da13f5/dev/preview/infrastructure/harvester/cert.tf#L19